src/loaders/JsonImageInfoLoader.ts

```diff
@@ -129,17 +137,22 @@ class JsonImageInfoLoader implements IVolumeLoader {
   }
 
   async loadVolumeData(vol: Volume, explicitLoadSpec?: LoadSpec, onChannelLoaded?: PerChannelCallback): Promise<void> {
-    const loadSpec = explicitLoadSpec || vol.loadSpec;
     // if you need to adjust image paths prior to download,
     // now is the time to do it.
     // Try to figure out the urlPrefix from the LoadSpec.
     // For this format we assume the image data is in the same directory as the json file.
+    const loadSpec = explicitLoadSpec || vol.loadSpec;
+    if (this.time !== loadSpec.time) {
+      this.time = loadSpec.time;
+      this.jsonInfo = await this.getJsonImageInfo(loadSpec);
+    }
     // This regex removes everything after the last slash, so the url had better be simple.
-    const urlPrefix = loadSpec.url.replace(/[^/]*$/, "");
-    this.imageArray.forEach((element) => {
-      element.name = urlPrefix + element.name;
-    });
-    JsonImageInfoLoader.loadVolumeAtlasData(vol, this.imageArray, onChannelLoaded);
+    const urlPrefix = this.urls[this.time].replace(/[^/]*$/, "");
+    const images = this.jsonInfo?.images.map((element) => ({ ...element, name: urlPrefix + element.name }));
```

Comment:
Might be good to remove any trailing slashes before doing this?
ex: `https://some-example.com/json_file.json/`